### ISSUE TITLE
fatal error in openbmc.pm terminates `nodeset <node> next` processing when node status switching from "installing" to "booting"

### ISSUE THREAD

**Issue by immarvin** [xcat2/xcat-core#3979]
steps to reproduce:
1. run xcatd with `xcatd -f` on mn
2. on cn:
````
cd /xcatpost
./updateflag.awk <mn ip> 3002
````
3. from the MN console same with step #1:
````
...
monitorctrl::setNodeStatusAttributes called
HTTP::Async object destroyed but still in use at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 460.
HTTP::Async INTERNAL ERROR: 'id_opts' not empty at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 460.
Use of uninitialized value in hash element at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 477.
Use of uninitialized value in hash element at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 477.
Use of uninitialized value in hash element at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 1179.
Use of uninitialized value in subroutine entry at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 1179.
Use of uninitialized value in subroutine entry at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 1179.
Can't use string ("") as a subroutine ref while "strict refs" in use at /opt/xcat/lib/perl/xCAT_plugin/openbmc.pm line 1179.

Died at /opt/xcat/sbin/xcatd line 2010.

xcatd: possible BUG encountered by xCAT install monitor service: Died at /opt/xcat/sbin/xcatd line 2010.
````

which prevents the following process of changing petitboot configuration

**Comment by zet809** [comment]
Hi, @immarvin , this issue is fixed with PR #3983 , the test looks like this, so will you pls close this issue? Thx!
```
# cat /tftpboot/petitboot/mid05tor12cn05
#install rhels7.4-snap2-ppc64le-compute
default xCAT
label xCAT
	kernel http://172.12.253.27:80/tftpboot/xcat/osimage/rhels7.4-snap2-ppc64le-install-compute/vmlinuz
	initrd http://172.12.253.27:80/tftpboot/xcat/osimage/rhels7.4-snap2-ppc64le-install-compute/initrd.img
	append "quiet inst.repo=http://172.12.253.27:80/install/rhels7.4-snap2/ppc64le inst.ks=http://172.12.253.27:80/install/autoinst/mid05tor12cn05 BOOTIF=70:e2:84:14:28:0e  ip=dhcp "
# ssh mid05tor12cn05
Last login: Thu Sep 21 02:50:32 2017 from gateway
[root@mid05tor12cn05 ~]# cd /xcatpost/
[root@mid05tor12cn05 xcatpost]# ./updateflag.awk 172.12.253.27 3002
[root@mid05tor12cn05 xcatpost]# exit
logout
Connection to mid05tor12cn05 closed.
# cat /tftpboot/petitboot/mid05tor12cn05
#boot
```

**Comment by robin2008** [comment]
We need test the concurrent case for operate the same openbmc node in future.
